protocol/x/vault/keeper/orders_test.go

```diff
@@ -249,79 +249,167 @@ func TestGetVaultClobOrders(t *testing.T) {
+			// 2. leverage = open_notional / equity
+			// 3. leverage_i = leverage +/- i * order_size_pct (- for ask and + for bid)
+			// 4. skew_i = -leverage_i * spread * skew_factor
+			// 5. a_i = oracle_price * (1 + skew_i) * (1 + spread)^{i+1}
+			//    b_i = oracle_price * (1 + skew_i) / (1 + spread)^{i+1}
+			// 6. subticks needs to be a multiple of subticks_per_tick (round up for asks, round down for bids)
+			// To calculate size of each order
+			// 1. `order_size_ppm * equity / oracle_price`.
 			expectedOrderSubticks: []uint64{
-				// spreadPpm = max(3_000, 1_500 + 4_200) = 5_700
-				// priceSubticks = 3_000_000_000 * 10^(-6 - (-9) + (-9) - (-6)) = 3 * 10^9
-				// a_1 = 3 * 10^9 * (1 + 0.0057)^1 = 3_017_100_000
-				3_017_100_000,
-				// b_1 = 3 * 10^9 * (1 - 0.0057)^1 = 2_982_900_000
-				2_982_900_000,
-				// a_2 = 3 * 10^9 * (1 + 0.0057)^2 = 3_034_297_470
-				// round up to nearest multiple of subticksPerTick=1000.
-				3_034_298_000,
-				// b_2 = 3 * 10^9 * (1 - 0.0057)^2 = 2_965_897_470
-				// round down to nearest multiple of subticksPerTick=1000.
-				2_965_897_000,
+				// spreadPpm = max(3_000, 4_500 + 50) = 4_550
+				// spread = 0.00455
+				// open_notional = -1_000 * 10^-9 * 3_000 * 10^6 = -3_000
+				// leverage = -3_000 / (2_000_000_000 - 3_000) = -3/1_999_997
+				// oracleSubticks = 3_000_000_000 * 10^(-6 - (-9) + (-9) - (-6)) = 3 * 10^9
+				// leverage_0 = leverage - 0 * 0.05 = -3/1_999_997
+				// skew_0 = 3 / 1_999_997 * 0.00455 * 0.7
+				// a_0 = 3 * 10^9 * (1 + skew_0) * (1 + 0.00455)^1 = 3_013_650_014.4
+				// round up to nearest multiple of subticks_per_tick=1_000.
+				3_013_651_000,
+				// b_0 = 3 * 10^9 * (1 + skew_0) / (1 + 0.00455)^1 = 2_986_411_840.46
+				// round down to nearest multiple of subticks_per_tick=1_000.
+				2_986_411_000,
+				// leverage_1 = leverage - 1 * 0.05
+				// skew_1 = -leverage_1 * 0.00455 * 0.7
+				// a_1 = 3 * 10^9 * (1 + skew_1) * (1 + 0.00455)^2 ~= 3_027_844_229.378
+				// round up to nearest multiple of subticks_per_tick=1_000.
+				3_027_845_000,
+				// leverage_1 = leverage + 1 * 0.05
+				// skew_1 = -leverage_1 * 0.00455 * 0.7
+				// b_1 = 3 * 10^9 * (1 + skew_1) / (1 + 0.00455)^2 ~= 2_972_411_780.773
+				// round down to nearest multiple of subticks_per_tick=5.
+				2_972_411_000,
+				// leverage_2 = leverage - 2 * 0.05
+				// skew_2 = -leverage_2 * 0.00455 * 0.7
+				// a_2 = 3 * 10^9 * (1 + skew_2) * (1 + 0.00455)^3 ~= 3_042_105_221.627
+				// round up to nearest multiple of subticks_per_tick=1_000.
+				3_042_106_000,
+				// leverage_2 = leverage + 2 * 0.05
+				// skew_2 = -leverage_2 * 0.00455 * 0.7
+				// b_2 = 3 * 10^9 * (1 + skew_2) / (1 + 0.00455)^3 ~= 2958477277.194
+				// round down to nearest multiple of subticks_per_tick=1_000.
+				2_958_477_000,
 			},
-			expectedOrderQuantums: []uint64{ // TODO (TRA-144): Implement order size
-				1000,
-				1000,
-				1000,
-				1000,
+			// order_size = 5% * 1999.997 / 3000 ~= 0.03333328333333334
```

Comment:
question: why is the equity slightly less than $2000 here?

Reply:
because of the -1000 base quantums perpetual position. Equity takes into account both asset positions (USDC) and perpetual positions. Is that as expected?

Reply:
im not sure I understand this. 
you could be short 1000 on ETH perp, but have even more equity since the current price is lower than your entry price.